pkg/userdata/flatcar/provider.go

```diff
@@ -33,6 +32,12 @@ import (
 	userdatahelper "github.com/kubermatic/machine-controller/pkg/userdata/helper"
 )
 
+const (
+	poseidonImage    = "quay.io/poseidon/kubelet"
+	hyperkubeImage   = "k8s.gcr.io/hyperkube-amd64"
+	lessThen118Check = "< 1.18"
```

Comment:
I'd use poseidon even for 1.17+ if it would be accessible because hyperkube was officially deprecated in 1.17 https://github.com/kubernetes/kubernetes/pull/85094, but poseidon is only consistently accessible since 1.18+.